src/test/java/de/fraunhofer/aisec/cpg/TestUtils.java

```diff
@@ -59,30 +59,74 @@
     return nodes.stream().filter(m -> m.getName().equals(name)).collect(Collectors.toList());
   }
 
-  public static List<TranslationUnitDeclaration> analyze(String fileExtension, Path topLevel)
+  /**
+   * Like {@link #analyze(List, Path, boolean, boolean)}, but for all files in a directory tree
+   * having a specific file extension
+   *
+   * @param fileExtension All files found in the directory must end on this String. An empty string
+   *     matches all files
+   * @param topLevel The directory to traverse while looking for files to parse
+   * @param usePasses Whether the analysis should run passes after the initial phase
+   * @param cleanupOnCompletion Whether {@link de.fraunhofer.aisec.cpg.graph.TypeManager} etc.
+   *     should be cleaned up after the analysis has ended
+   * @return A list of {@link TranslationUnitDeclaration} nodes, representing the CPG roots
+   * @throws Exception Any exception thrown during the parsing process
+   */
+  public static List<TranslationUnitDeclaration> analyze(
+      String fileExtension, Path topLevel, boolean usePasses, boolean cleanupOnCompletion)
       throws Exception {
-    File[] files =
+    List<File> files =
         Files.walk(topLevel, Integer.MAX_VALUE)
             .map(Path::toFile)
             .filter(File::isFile)
             .filter(f -> f.getName().endsWith(fileExtension))
             .sorted()
-            .toArray(File[]::new);
+            .collect(Collectors.toList());
+    return analyze(files, topLevel, usePasses, cleanupOnCompletion);
+  }
 
-    TranslationConfiguration config =
+  /**
+   * Default way of parsing a list of files into a full CPG. All default passes are applied
+   *
+   * @param topLevel The directory to traverse while looking for files to parse
+   * @param usePasses Whether the analysis should run passes after the initial phase
+   * @param cleanupOnCompletion Whether {@link de.fraunhofer.aisec.cpg.graph.TypeManager} etc.
+   *     should be cleaned up after the analysis has ended
+   * @return A list of {@link TranslationUnitDeclaration} nodes, representing the CPG roots
+   * @throws Exception Any exception thrown during the parsing process
+   */
+  public static List<TranslationUnitDeclaration> analyze(
+      List<File> files, Path topLevel, boolean usePasses, boolean cleanupOnCompletion)
+      throws Exception {
+    TranslationConfiguration.Builder builder =
         TranslationConfiguration.builder()
-            .sourceLocations(files)
+            .sourceLocations(files.toArray(File[]::new))
```

Comment:
Can we convert `sourceLocations` to a list as well? Why is it an array

Reply:
It has always been a File vararg. No idea whether this should be changed. I wouldn't mind

Reply:
I think I'll just add an overloading function. I just saw that it's converted back to a list anyway :sweat_smile:

Reply:
Ah, I didn't see that it's a varags, I think that's probably there for convenience. I avoid changing too much public API in this PR, so adding an overloaded function is ok as long as the old API still works.